WordPress/Classes/Utility/DisplayableImageHelper.m

```diff
@@ -109,6 +109,34 @@ + (NSString *)searchPostContentForImageToDisplay:(NSString *)content
     return imageSrc;
 }
 
++ (NSSet *)searchPostContentForAttachmentIdsInGalleries:(NSString *)content
+{
+    NSMutableSet *resultSet = [NSMutableSet set];
+    // If there is no gallery shortcode in the content, just bail.
+    if (!content || [content rangeOfString:@"[gallery "].location == NSNotFound) {
+        return resultSet;
+    }
+
+    // Get all the things
+    static NSRegularExpression *regexGallery;
+    static dispatch_once_t onceTokenRegexGallery;
+    dispatch_once(&onceTokenRegexGallery, ^{
+        NSError *error;
+        NSString *galleryPattern = @"\\[gallery[^]]+ids=\"([0-9,]*)\"[^]]*\\]";
+        regexGallery = [NSRegularExpression regularExpressionWithPattern:galleryPattern options:NSRegularExpressionCaseInsensitive error:&error];
+    });
+
+    // Find all the gallery shortcodes in the content passed.
+    NSArray *matches = [regexGallery matchesInString:content options:0 range:NSMakeRange(0, [content length])];
+
+    for (NSTextCheckingResult *match in matches) {
+        NSString *tag = [content substringWithRange:match.range];
```

Comment:
Cool. So with that updated regex, the NSTextCheckingResult will now return multiple ranges.  The first range, the one referenced by the `match.range` property is for the entire match.  However, we now have a match on just the gallery IDs via `match.rangeAtIndex(1)`.  So we can use that to avoid a lot of the string parsing and other work in `idsFromGallery`.   A very rough example would be: 
```
        NSRange rng = [match rangeAtIndex:1];
        NSString *tag = [content substringWithRange:rng];
        NSArray *arr = [tag componentsSeparatedByString:@","];
        NSMutableArray *idsArr = [NSMutableArray array];
        for (NSString *item in arr) {
            [idsArr addObject:[item numericValue]];
        }
        NSSet *tagIds = [NSSet setWithArray:idsArr];
```